### PR TITLE
Improve ZerosLike implementation and optimize for opset >= 9

### PR DESCRIPTION
**Pull request by hwangdeyu**
Replace `cast + mul(value 0) + cast` to `shape + ConstantShape(value 0)` when opset >= 9 [ConstantShape](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ConstantOfShape-9), which can get a better inference performance.

fixes https://github.com/onnx/tensorflow-onnx/issues/1997

### CODE REVIEW

File: tf2onnx/onnx_opset/generator.py [hunk 1]
```diff
@@ -242,6 +242,16 @@ def version_1(cls, ctx, node, **kwargs):
                       name=node.name, outputs=node.output,
                       shapes=shapes, dtypes=dtypes)
 
+    @classmethod
+    def version_9(cls, ctx, node, **kwargs):
+        dtypes = node.output_dtypes
+        ctx.remove_node(node.name)
+        shape = ctx.make_node("Shape", node.input).output[0]
+        zero_tensor = helper.make_tensor("value", dtypes[0], [1], vals=[0])
```
Comment (hwangdeyu) [inline tf2onnx/onnx_opset/generator.py:250]: Make `ConstantOfShape` to `0`.

File: tf2onnx/onnx_opset/generator.py [hunk 2]
```diff
@@ -242,6 +242,16 @@ def version_1(cls, ctx, node, **kwargs):
...
+        ctx.make_node("ConstantOfShape", inputs=[shape],
+                      attr={'value': zero_tensor},
+                      name=node.name, outputs=node.output,
+                      dtypes=dtypes)
```
Comment (fatcat-z) [inline tf2onnx/onnx_opset/generator.py:254]: Please add one more blank line after this method.

Comment (hwangdeyu) [reply tf2onnx/onnx_opset/generator.py:254]: Sure, added.

**Review by fatcat-z — APPROVED** [review]
LGTM, thanks!

**Pull request merged**